configure

```diff
@@ -13,16 +17,17 @@ del _
 
 import sys
 from distutils.spawn import find_executable as which
-if sys.version_info[0] != 2 or sys.version_info[1] not in (6, 7):
-  sys.stderr.write('Please use either Python 2.6 or 2.7')
-
-  python2 = which('python2') or which('python2.6') or which('python2.7')
-
-  if python2:
-    sys.stderr.write(':\n\n')
-    sys.stderr.write('  ' + python2 + ' ' + ' '.join(sys.argv))
 
+current_python = 'python{0}.{1}'.format(*sys.version_info)
+valid_pythons = ('python3.7', 'python3.6', 'python3.5', 'python2.7')
+if current_python in valid_pythons:
+  import configure
+else:
+  sys.stderr.write('Please use '+ ', '.join(valid_pythons))
+  # try to find a locally installed acceptable python
+  for py in valid_pythons:
+    python_exe = which(py)
+    if python_exe:
+      sys.stderr.write(':\n\n  {0} {1}'.format(python_exe, ' '.join(sys.argv)))
+      break
   sys.stderr.write('\n')
-  sys.exit(1)
```

Comment:
Don't we want a non-zero exit code on failure?

Reply:
Great catch!  Thanks.